pkg/procstats/procstats.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019 Intel Corporation. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package procstats
+
+import (
+	"io/ioutil"
+	"strconv"
+	"strings"
+	"sync"
+)
+
+// CPUTimeStat is used to calculate the CPU usage.
+type CPUTimeStat struct {
+	sync.RWMutex
+	PrevIdleTime       []uint64
+	PrevTotalTime      []uint64
+	CurIdleTime        []uint64
+	CurTotalTime       []uint64
+	DeltaIdleTime      []uint64
+	DeltaTotalTime     []uint64
+	CPUUsage           []float64
+	IsGetCPUUsageBegin bool
+	CPUNumber          int
+}
+
+var (
+	// procRoot is the mount point for the proc filesystem
+	procRoot = "/proc"
+	procStat = procRoot + "/stat"
+)
+
+// GetCPUTimeStat calculates CPU usage by using the CPU time statistics from /proc/stat
+func (t *CPUTimeStat) GetCPUTimeStat() error {
+	lines, err := readLines(procStat)
+	if err != nil {
+		return err
+	}
+	// /proc/stat looks like this:
+	// cpuid：user，nice, system, idle, iowait, irq, softirq
```

Comment:
Line 51: the colon and the first comma are some special non-ASCII characters after `cpuid` and `user`. These should be fixed.